src/c_wrapper.jl

```diff
@@ -16,28 +23,64 @@ mutable struct ScsMatrix{T} <: AbstractSCSType
     n::T
 end
 
+function ScsMatrix(A::ManagedScsMatrix)
```

Comment:
Ditto on this method. We shouldn't overload constructors of the `ScsXXX` types. I'd also prefer not to have the `ManagedScsMatrix` type, and just store the individual vectors in the `_ScsDataWrapper` struct.

Reply:
that will be 6 vectors + 2 pointers; I went this path to have 2 + 2;

at this point it seems that we pushed the unwieldiness of `GC.@preserve` to the constructor of `_ScsDataWrapper` ;)

Reply:
> it seems that we pushed the unwieldiness of GC.@preserve to the constructor of _ScsDataWrapper

Yes. 

I've generally found that after managing all the different solver wrappers, the less intermediate calls you have the better. Although it's nice to have these helper functions, we don't expect users to call them, so we should be as simple and explicit as possible. These redirections just cause later confusion when you have to reason about new types (`ManagedScsMatrix`) and new constructors (`ScsMatrix`), as opposed to a single type (`_ScsDataWrapper`) with a single---albeit unwieldy---constructor.